src/User/AutoLoginMiddleware.php

```diff
@@ -1,16 +1,81 @@
 <?php
 
+declare(strict_types=1);
+
 namespace Yiisoft\Yii\Web\User;
 
+use Psr\Http\Server\RequestHandlerInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Yiisoft\Auth\IdentityRepositoryInterface;
+use Yiisoft\Yii\Web\User\User;
+
 /**
  * AutoLoginMiddleware automatically logs user in based on "remember me" cookie
  */
-class AutoLoginMiddleware
+final class AutoLoginMiddleware implements MiddlewareInterface
 {
     private User $user;
+    private IdentityRepositoryInterface $identityRepository;
 
-    public function __construct(User $user)
-    {
+    public function __construct(
+        User $user,
+        IdentityRepositoryInterface $identityRepository
+    ) {
         $this->user = $user;
+        $this->identityRepository = $identityRepository;
+    }
+
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
+    {
+        if (!$this->userIsAuth($request)) {
+            throw new \Exception('Error authentication');
```

Comment:
This behavior will prevent a user with invalid cookie from viewing a website at all. I think we need to behave as in Yii 2 in this case i.e. logging an error: https://github.com/yiisoft/yii2/blob/master/framework/web/User.php#L602

Reply:
This I don't Understand. What is the change?. Thanks!

Reply:
1. Require `Psr\Log\LoggerInterface` in constructor.
2. Do `$this->logger->warning('Unable to authenticate used by cookie.');` instead of throwing exception.

Reply:
Great, updated in the last commit.